### PR TITLE
simplify framework selection

### PR DESCRIPTION
**Pull request by omsmith**

### CODE REVIEW

File: csharp/private/providers.bzl [hunk 1]
```diff
@@ -81,77 +81,54 @@ CSharpAssembly = {
...
+#
+# https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support
+#   4.7.2 or higher.
+#   we recommend that you upgrade the project to target .NET Framework
+#   projects. For .NET Framework projects that need to use such libraries,
+#   libraries that were built for those versions from .NET Framework 4.6.1
+#   1.5 through 2.0, there are several issues with consuming .NET Standard
+#   While NuGet considers .NET Framework 4.6.1 as supporting .NET Standard
+# NOTE: Microsoft has this to say:
+FrameworkCompatibility["netstandard1.4"] = ["netstandard1.5", "net461"] + FrameworkCompatibility["netstandard1.5"] + FrameworkCompatibility["net461"]
```
Comment (omsmith) [inline csharp/private/providers.bzl]: As-is, these arrays currently have a bunch of duplicates

File: csharp/private/providers.bzl [hunk 2]
```diff
@@ -81,77 +81,54 @@ CSharpAssembly = {
...
-    "netstandard1.1": 4,
-    "net45": 5,
-    "netstandard1.2": 6,
-    "net451": 7,
-    "net452": 8,
-    "netstandard1.3": 9,
-    "net46": 10,
+# A dict of target frameworks to the list of other target frameworks that can
+# compile against it
+FrameworkCompatibility = {}
```
Comment (omsmith) [inline csharp/private/providers.bzl]: This would benefit from being generated

File: csharp/private/common.bzl [hunk 3]
```diff
@@ -105,16 +47,19 @@ def fill_in_missing_frameworks(providers):
       deps: the original deps used to generate the providers.
     """
 
-    for gap_info in _find_framework_gaps(providers):
-        (base, compat_set_idx, start_idx, end_idx) = gap_info
+    # iterate through the compatability table since it's in preference order
```
Comment (omsmith) [inline csharp/private/common.bzl]: This is fine as long as `netcoreapp` doesn't start supporting `net` or vice-versa

File: csharp/private/common.bzl [hunk 4]
```diff
@@ -105,15 +47,17 @@ def fill_in_missing_frameworks(providers):
...
+    for tfm in FrameworkCompatibility.keys():
```
Comment (j3parker) [inline csharp/private/common.bzl]: Haha this is even nicer than we talked about (doing 3 loops for each family.) Nice!

**Pull request merged**